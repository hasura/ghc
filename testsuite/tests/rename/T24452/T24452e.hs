-- Multiple unrelated errors related to empty.
{-# LANGUAGE DuplicateRecordFields #-}
{-# LANGUAGE OverloadedRecordDot #-}
{-# LANGUAGE OverloadedRecordUpdate #-}

module T24452e where

import Control.Applicative (Alternative)
import qualified Data.Map as Map
import qualified Data.Set as Set

data A = A {
    empty :: ()
}
data B = B {
    empty :: ()
}

foo = empty

newtype Foo a = MkFoo [a] deriving (Functor, Applicative)

instance Alternative Foo where
  empty = undefined