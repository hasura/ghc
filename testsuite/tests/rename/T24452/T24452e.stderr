T24452e.hs:19:7: error: [GHC-87543]
    Ambiguous occurrence ‘empty’.
    It could refer to
       either the field ‘empty’ of record ‘A’,
              defined at T24452e.hs:13:5,
           or the field ‘empty’ of record ‘B’,
              defined at T24452e.hs:16:5.

T24452e.hs:24:3: error: [GHC-54721]
    ‘empty’ is not a (visible) method of class ‘Alternative’